tools/type_whisperer/BUILD

```diff
@@ -77,6 +77,15 @@ file_descriptor_set_text(
     deps = ["@envoy_api_canonical//:all_protos"],
 )
 
+file_descriptor_set_text(
+    name = "all_protos_with_ext_pb_text_inc_source",
+    with_external_deps = True,
+    deps = [
+        "//source:protos",
```

Comment:
Why do we need these in the type DB again?

Reply:
without this it was failing to validate the `SimpleHttpCache` config which is specified in the source protos

Reply:
I think configs in `//source/` or `//test/` don't need strict validation; they are not consumed by any external system and we can simplify corner cases here.

Reply:
> I think configs in //source/ or //test/ don't need strict validation

they are not being validated.

the protos in //source are being *used* for validation

Reply:
there are no configs in //test just some certs that are used in the other (non-protobuf) validation

Reply:
Is this only about `envoy.source.extensions.filters.http.cache.SimpleHttpCacheConfig`? If so, can you just scope to that one proto. It's almost technically a part of the API, but none of the other protos in `//source` should be I think.

Reply:
so none of the others listed here are significant - https://github.com/envoyproxy/envoy/pull/15284#issuecomment-790951767 ?

im wondering why some (or maybe only one) of the protos required to validate are not in the api tree

im also concerned that we can validate properly - afaict the `SimpleHttpCacheConfig` is valid and therefore should be validated  - is this really the only one not properly in the api tree ?

Reply:
> A quick grep for envoy\.source seems to indicate that SimpleHttpCacheConfig is the only one.

can/should we move it ?

ill update the `_inc_source` target now to only include that proto...